docs/theming/developers.md

```diff
@@ -21,42 +22,185 @@ order: 4
   import '@rhds/elements/rh-button/rh-button.js';
   import '@rhds/elements/rh-card/rh-card.js';
+  import '@rhds/elements/rh-code-block/rh-code-block.js';
   import '@rhds/elements/rh-cta/rh-cta.js';
   import '@rhds/elements/rh-pagination/rh-pagination.js';
   import '@rhds/elements/rh-tabs/rh-tabs.js';
   import '@rhds/elements/rh-tag/rh-tag.js';
   import '@rhds/elements/rh-tile/rh-tile.js';
 </script>
 
-## Context Protocol
-
-## Style Queries
-
-Your page markup declares a custom surface:
-
-```html
-<div class="custom-surface dark">
-  <rh-cta href="#">GO!</rh-cta>
-</div>
-```
-
-And your document css sets the desired color context:
-
-```css
-.custom-surface {
-  container: surface;
-  &.dark { --rh-background-context: dark; }
-  &.light { --rh-background-context: light; }
-}
-```
-
-Internally / in shared CSS:
-
-```css
-@container style (--rh-context-background: dark) {
-  --rh-color-text-primary: var(--rh-color-text-primary-on-dark)
-}
-```
+## Current implementation
+
+RHDS' Theming system is primarily about styles, but it currently relies on JavaScript to work.
+
+### Context protocol
+
+The context protocol is an [open protocol][contextprotocol] developed by the web
+components community to support the passing of data between components.
+
+Our system utilizes this protocol with the setting of the `color-palette`
+attribute on a provider element which passes it's context data of `light` or
+`dark` on to it's children.  By doing so we can ensure accessible colors are
+applied given any possible change in context value higher up in the DOM tree. 
+
+This is important not only for helping us ensure we maintain our own design
+guidelines, but also for accessibility compliance and enabling great experiences
+for all our users.
+
+The context protocol is enabled by two [reactive controllers][controllers]: the
+[provider controller][providersrc], and the [consumer controller][consumersrc].
+These controllers work in tandem to alternately provide the context data, or
+consume it as a child element. Custom elements can implement the consumer, the
+provider, or both, depending on the needs of that particular element.
+
+#### Providers
+
+Custom Elements that implement the provider controller are elements that provide
+their own context, overriding that of their parent.  Elements such as
+`<rh-surface>`, `<rh-card>`, and `<rh-accordion>` are examples of such context
+providers. If a provider contains a set `color-palette` attribute, it will
+override any parent context, and pass it's context on to its children.
+
+To make your element a color context provider,
+
+1. First import the provider controller.
+  <rh-code-block dedent language="js" highlighting="client">
+    <script type="sample/javascript">
+      import { colorContextProvider, type ColorPalette } from '../../lib/context/color/provider.js';
+    </script>
+  </rh-code-block>
+2. Then add the `@colorContextProvider` decorator to a property with the
```

Comment:
Looks like number two is not indenting properly [in the DP](https://deploy-preview-1897--red-hat-design-system.netlify.app/theming/developers/#providers).